pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -266,3 +269,33 @@ func (p *ConfigurationEditFlags) AnyMutation(cmd *cobra.Command) bool {
 	}
 	return false
 }
+
+func envVarsFromSlice(envFlags []string, delimiter string) ([]corev1.EnvVar, error) {
+	envDict := make(map[string]string, len(envFlags))
+	envFlagsUpdated := envFlags[:0]
+	for _, flag := range envFlags {
+		envPair := strings.SplitN(flag, delimiter, 2)
+		switch len(envPair) {
+		case 0:
+			return nil, fmt.Errorf("argument requires a value that contains the %q character; got %q", delimiter, envPair)
+		case 1:
+			if _, ok := envDict[envPair[0]]; !ok {
```

Comment:
IIUC, you added this check to ensure if user gave same env var multiple times.
The envDict though uses the value of the env parsed last.

Reply:
> The envDict though uses the value of the env parsed last.

I would like to use the value of last parsed env vars, however regarding the order it makes sense to use the first timing to be appeared in the user input, because I have an image of updating the value with the same key multiple times in case that the user specified duplicate pair of env vars. In addition, it becomes a little complicated to update the order based on the count of duplicate entry. WDYT?